docs/rfcs/XeTile.md

```diff
@@ -0,0 +1,126 @@
+```mlir
+  %tile_updated = XeTile.update_tile_offset %tile, %offset_x, offset_y :
+		tile<64x64xbf16>, index, index into tile <64x64xbf16>
+```
+
+`xetile.wg_map` mapping attribute allows XeTile operation to work at the workgroup level. Without these attributes, the XeTile works at the subgroup level. With wg_map attributes, XeTile operations can be applied to workgroup-level tile sizes. The attribute `xetile.wg_map` guide the lowering from the workgroup level to the subgroup level by specifying how the data is distributed across parallel subgroups. 
+`xetile.sg_map` attributes allows the user to further specify the mapping of each data element to each work item thread. It works the same way as `xegpu.sg_map` defined in XeGPU dialect. 
+`xetile.wg_map` and `xeTile.sg_map` maps give the user full control over the lowering process so that the user can tune the tiling size for both the workgroup and subgroup to tune the performance.
+
+Below is an example.
+```mlir
+   #wg_map_a = #xetile.wg_map<sg_layout = [2, 2], sg_data = [32, 128]>
+   #sg_map_a = #xetile.sg_map<wi_layout = [2, 8], wi_data = [1, 2]>
+   #xe_map_a = #xetile.xe_map<wg = #wg_map_a, sg = #sg_map_a>
+
+   %a_init_tile = xetile.init_tile %A[%m, %c0] : memref<1024x1024xf16> -> !xetile.tile<128x128xf16, #xe_map_a>
+```
+Within the `xetile.wg_map`, `sg_layout` specifies the subgroup layout, and `sg_data` specifies the tile size owned by each subgroup. In the example above, sg_layout [2,2] means that each workgroup has 4 subgroups with 2 rows and 2 columns. sg_data [32,128] means that each subgroup works on a submatrix [32, 128]. The data elements assigned to each subgroup thread must be contiguous.
+
+For each dimension, the tile size must be divisible by `sg_data`, and the size of `sg_layout` multiplying `sg_data` must be divisible by the tile size. The tile is distributed to sg_data in a round-robin fashion. When the tile size is smaller than the size of sg_layout multiplying sg_data, the tile data is wrapped around and continues to be distributed to subgroup threads. For example, for the tile size [128, 128] and sg_data [32, 128], along the second dimension, there is no more data left to assign after the first subgroup, it wraps around and moves to the beginning of the tile and continues the assignment. Instead, for the first dimension, there is more data left after the first round of distribution, so it move to the next subtile and continue the assignement. As a result, the tile would be sliced to four subtiles with size [32,128], with the following mapping: 
+
```

Comment:
the size of `sg_layout` multiplying `sg_data` must be divisible by the tile size. 
what do you mean by this?
for A, [32,128]x[2,2] = [64,256] is not divisible by [128,128]
for B, [128, 32]x[2,2] = [256,64] is not divisible by [128, 128]
for C, [32,32]x[2,2] = [64, 64] the reverse is true.

I would prefer to say this,

for the C tile sg_layout x sg_data must evenly divide the C wg level tile size. For A and B this does not have a meaning in my opinion.

Reply:
Reword,
For each dimension, the tile size must be divisible by `sg_data`

to 

value in each dim of the sg_data must evenly divide the corresponding value in the wg_tile. For example, 128%128 = 0 and 128%32 = 0 for sg_data = [128, 32] and wg_tile = [128, 128].

Reply:
Data distribution among the subgroups work as follows. If sg_data[i] x sg_layout[i] < wg_tile[i] the we have enough data to distribute to the next subgroup along dimension i. So the next slice is assigned to (i+1)'th subgroup in that dimension. If sg_data[i] x sg_layout[i] >= wg_tile[i] we have already used up all the data along that dimension. In that case we wrap around the tile and assign the same tile to (i+1)'th subgroup. For example,....